File: classified/packages/server/src/db/schema/serverAgent.ts
import { pgTable, uuid, primaryKey } from 'drizzle-orm/pg-core';
import { messageServerTable } from './messageServer';

export const serverAgentsTable = pgTable(
  'server_agents',
  {
    serverId: uuid('server_id')
      .notNull()
      .references(() => messageServerTable.id, { onDelete: 'cascade' }),
    agentId: uuid('agent_id').notNull(), // No foreign key to agent table since agents are managed elsewhere
  },
  (table) => ({
    pk: primaryKey({ columns: [table.serverId, table.agentId] }),
  })
);

File: cpp_merge/eliza/packages/plugin-sql/src/schema/serverAgent.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/eliza/packages/plugin-sql/src/schema/serverAgent.h"

any serverAgentsTable = pgTable(std:("server_agents"), object{
    object::pair{std:("serverId"), uuid(std:("server_id"))->notNull()->references([=]() mutable
    {
        return messageServerTable->id;
    }
    , object{
        object::pair{std:("onDelete"), std:("cascade")}
    })}, 
    object::pair{std:("agentId"), uuid(std:("agent_id"))->notNull()->references([=]() mutable
    {
        return agentTable->id;
    }
    , object{
        object::pair{std:("onDelete"), std:("cascade")}
    })}
}, [=](auto table) mutable
{
    return (object{
        object::pair{std:("pk"), primaryKey(object{
            object::pair{std:("columns"), array<any>{ table["serverId"], table["agentId"] }}
        })}
    });
}
);

void Main(void)
{
}

MAIN
